src/app/shared/table-of-contents/table-of-contents.ts

```diff
@@ -104,7 +104,7 @@ export class TableOfContents implements OnInit, AfterViewInit, OnDestroy {
     this._links = [];
   }
 
-  addHeaders(sectionName: string, docViewerContent: HTMLElement) {
+  addHeaders(sectionName: string, docViewerContent: HTMLElement, sectionIndex = 0) {
```

Comment:
I have a slight preference for making the `sectionIndex` the 2nd argument rather than the 3rd in both of these functions.